test/cpp/end2end/xds/xds_end2end_test_lib.h

```diff
@@ -193,7 +193,7 @@ class XdsEnd2endTest : public ::testing::TestWithParam<XdsTestType> {
   // Default values for locality fields.
   static const char kDefaultLocalityRegion[];
   static const char kDefaultLocalityZone[];
-  static const int kDefaultLocalityWeight = 3;
+  static const uint32_t kDefaultLocalityWeight = 3;
```

Comment:
Is this 32 bit? Also, why this change? If we are making this change, shouldn't we be consistent and change priority too?

Reply:
This matches the definition of the field in the xDS protos.

The reason for making this change for weight is that we realized that there's a risk of integer overflow when we aggregate the weights in the weighted_target policy (which is what this PR fixes).  We don't ever aggregate priorities, so there's no need to change the type we're using there.